src/cider/nrepl/middleware/info.clj

```diff
@@ -280,23 +280,34 @@
                             (sort-by count))
     :else (:arglists info)))
 
-(defn format-eldoc [raw-eldoc]
-  (map #(mapv str %) raw-eldoc))
+(defn format-arglists [raw-arglists]
+  (map #(mapv str %) raw-arglists))
+
+(defn extract-ns
+  [info]
+  (cond
+    (:ns info) (:ns info)
+    (:candidates info) (->> (:candidates info)
+                            keys)
+    :else (:class info)))
+
+(defn format-ns
+  [raw-ns]
+  (if (coll? raw-ns)
+    (clojure.string/join ", " (map str raw-ns))
+    (str raw-ns)))
 
 (defn eldoc
   [info]
-  (if-let [raw-eldoc (extract-eldoc info)]
-    (format-eldoc raw-eldoc)))
+  (if-let [arglists (extract-arglists info)]
+    {:eldoc (format-arglists arglists)
+     :name (:name info)
+     :ns (-> info extract-ns format-ns)}
```

Comment:
We could also return a `:class` key, to differentiate between classes and ns'. Would that be better ?